### PR TITLE
runtests.sh optionally set path to python

### LINKED ISSUE #1036: runtests.sh fails
**Describe the bug**
`./runtests.sh --codeformat --coverage` gives the same error both on my local OSX and remote linux.

Output is:

```
/root/MONAI:/root/MONAI
Compiling and installing MONAI cpp extensions...
  File "setup.py", line 29
    print(f"setup.py with torch {torch.__version__}")
                                                   ^
SyntaxError: invalid syntax
```

**To Reproduce**
1. `cd to MONAI root`
2. `./runtests.sh --codeformat --coverage`

**Expected behavior**
Tests should run

**Environment (please complete the following information):**
 - OS: linux and OSX
 - Python version: 3.6.9 (linux) 3.8.5 (OSX)
 - MONAI version: master (fa521a9291405ed8a398a6f8a97ad5a8f7c46e96)

### PR DESCRIPTION
**Pull request by rijobro**
Fixes #1036.

### Description
option to set environmental variable `MONAI_PY_EXE` to set the python executable. If this environmental variable is not, then `$(which python)` is used as before, ensuring backwards compatibility. Future python calls then use `$PY_EXE`.

pip-installed packages are now run with e.g., `$PY_EXE -m flake8` instead of just `flake8`, in case they are not added to `PATH`. For example:

```
> $PY_EXE -m pip install -r requirements-dev.txt
> flake8
zsh: command not found: flake8
> $PY_EXE -m flake8
[command runs successfully]
```

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.

### CODE REVIEW

File: runtests.sh [hunk 1]
```diff
@@ -171,6 +174,10 @@ function print_style_fail_msg() {
     echo "Please run auto style fixes: ${green}./runtests.sh --autofix --nounittests${noColor}"
 }
 
+function is_pip_installed() {
+	return $($PY_EXE -c "import sys, pkgutil; sys.exit(0 if pkgutil.find_loader(sys.argv[1]) else 1)" $1)
```
Comment (wyli) [inline runtests.sh:178]: could change `$PY_EXE` to `${PY_EXE}`?

File: runtests.sh [hunk 2]
```diff
@@ -456,7 +463,7 @@ fi
 
 
 # testing command to run
-cmd="python3"
+cmd="${PY_EXE}"
```
Comment (wyli) [inline runtests.sh:466]: I think the `coverage` commands could also be `${PY_EXE} -m coverage`

Comment (rijobro) [reply runtests.sh:466]: done, thanks

**Review by wyli — APPROVED** [review]
thanks!

**Pull request merged**